plugins/react.js

```diff
@@ -11,6 +11,10 @@ module.exports = {
   parserOptions: {
     ecmaFeatures: {
       jsx: true
+    },
+    requireConfigFile: false,
+    babelOptions: {
+      presets: ['@babel/preset-react']
```

Comment:
We don't want this `requireConfigFile` setting here. The existence of the config file is generally desirable; we turn it off for the repos where it is known not to exist.

But this is raising a larger question for me... I had [verified](https://github.com/zendeskgarden/eslint-config/pull/176) the `@babel/eslint-parser` upgrade against both `react-components` and `react-containers`. I'm back to wondering if this is a JSX support preset that actually should live here?

After re-reading https://www.npmjs.com/package/@babel/eslint-parser re: `requireConfigFile`, I'm convinced that the majority case is that repos with React/JSX will also have a Babel config that the `@babel/eslint-parser` will follow. And `svg-icons` is the minority case that should have this `babelOptions: { presets: ... }` setting as a one-off.

A 1000 humble apologies for the whip around on this @mtomcal. But, as a centralized config for all of Garden, I think it is worth worrying about getting this right. tldr; I'm thinking we don't need this PR and should keep the added `babelOptions` config local to `svg-icons`.